### PR TITLE
feat(util): Add more verbose output to spawn process

### PR DESCRIPTION
**Pull request by HazAT**
/roast me

### CODE REVIEW

File: lib/targets/pods.js [hunk 1]
```diff
@@ -44,8 +44,12 @@ module.exports = async (context) => {
     context.logger.info(`Pushing podspec ${fileName} to cocoapods`);
     if (shouldPerform()) {
       const env = _.pick(process.env, ['COCOAPODS_TRUNK_TOKEN', 'PATH']);
-      await spawn(COCOAPODS_BIN, ['repo', 'update'], { cwd: directory, env });
-      await spawn(COCOAPODS_BIN, ['trunk', 'push', fileName], { cwd: directory, env });
+      try {
+        await spawn(COCOAPODS_BIN, ['repo', 'update'], { cwd: directory, env });
+        await spawn(COCOAPODS_BIN, ['trunk', 'push', fileName], { cwd: directory, env });
+      } catch (e) {
+        context.logger.error(`pod command failed with: ${e}`);
```
Comment (jan-auer) [inline lib/targets/pods.js]: Rethrow the exception here, otherwise the "success" message would be printed, too.

File: lib/utils.js [hunk 2]
```diff
@@ -66,16 +66,38 @@ function cloneContext(context, params = {}) {
 /**
  * Creates an error object and attaches the error code
  *
- * @param {number} code A non-zero error code
+ * @param {number} code
```
Comment (jan-auer) [inline lib/utils.js]: Missing doc

File: lib/utils.js [hunk 3]
```diff
@@ -66,16 +66,38 @@ function cloneContext(context, params = {}) {
...
+ *
+ * @param {string[]} args Optional arguments to pass to the command
+ */
+function convertOptionsForError(options) {
+  let envVars = {};
+  if (options && options.env) {
+    envVars = { envVars: Object.keys(options.env) };
+    delete options.env;
+  }
+  return JSON.stringify(Object.assign({}, options, envVars));
```
Comment (kamilogorek) [inline lib/utils.js]: Try `JSON.stringify(Object.assign({}, options, envVars), null, 2)` and see some magic.

```js
JSON.stringify(foo)
// "{"foo":1,"bar":[1,2,3],"baz":{"quz":[42]}}"
```

```js
JSON.stringify(foo, null, 2)
"{
  "foo": 1,
  "bar": [
    1,
    2,
    3
  ],
  "baz": {
    "quz": [
      42
    ]
  }
}"
```

File: lib/utils.js [hunk 4]
```diff
@@ -66,16 +66,38 @@ function cloneContext(context, params = {}) {
...
 /**
+ * Strips env values from the options array
+ *
+ * @param {string[]} args Optional arguments to pass to the command
+ */
+function convertOptionsForError(options) {
+  let envVars = {};
+  if (options && options.env) {
+    envVars = { envVars: Object.keys(options.env) };
+    delete options.env;
```
Comment (jan-auer) [inline lib/utils.js]: Do not mutate options, just override `env`

File: lib/utils.js [hunk 5]
```diff
@@ -66,16 +66,38 @@ function cloneContext(context, params = {}) {
...
+ * @param {string} command The command to run
+ * @param {string[]} args Optional arguments to pass to the command
+ * @param {object} options Optional options to pass to child_process.spawn
  * @returns {Error} The error with code
  */
-function exitCodeError(code) {
-  const error = new Error(`Process exited with code ${code}`);
+function exitCodeError(code, command, args, options) {
+  const error = new Error(
+    `Process exited with code ${code}\n` +
```
Comment (jan-auer) [inline lib/utils.js]: more concise output to copy paste to shell

File: lib/utils.js [hunk 6]
```diff
@@ -86,9 +108,15 @@ function exitCodeError(code) {
  */
 function spawn(command, args, options) {
   return new Promise((resolve, reject) => {
-    const process = child.spawn(command, args, options);
-    process.on('exit', code => (code === 0 ? resolve() : reject(exitCodeError(code))));
-    process.on('error', reject);
+    try {
+      const process = child.spawn(command, args, options);
+      process.on('exit', code => (code === 0 ? resolve() : reject(exitCodeError(code, command, args, options))));
+      process.on('error', error => {
```
Comment (jan-auer) [inline lib/utils.js]: one liner

File: lib/utils.js [hunk 7]
```diff
@@ -86,9 +108,15 @@ function exitCodeError(code) {
...
+        reject(exitCodeError(error.code, command, args, options));
```
Comment (jan-auer) [inline lib/utils.js]: `error.code` is not documented. Does that really exist?

File: lib/utils.js [hunk 8]
```diff
@@ -86,9 +108,15 @@ function exitCodeError(code) {
...
+      });
+    } catch (error) {
+      reject(exitCodeError(error.code, command, args, options));
```
Comment (jan-auer) [inline lib/utils.js]: Same here

**Comment by jan-auer** [comment]
@kamilogorek would you mind reviewing my changes as well?

File: lib/utils.js [hunk 9]
```diff
@@ -64,13 +64,32 @@ function cloneContext(context, params = {}) {
 }
 
 /**
+ * Strips env values from the options object
+ *
+ * @param {object} options Optional options passed to spawn
+ */
+function convertOptionsForError(options = {}) {
+  const override = options.env && { env: Object.keys(options.env) };
```
Comment (kamilogorek) [inline lib/utils.js]: It will break if you pass `null` as `options`, but it's up to you if you want to cover this edge-case

Comment (jan-auer) [reply lib/utils.js]: good eye, but let's take that risk for now :)

**Pull request merged**